### ISSUE TITLE
copyq fails with too many files open

### ISSUE THREAD

**Issue by aplatypus** [hluk/CopyQ#1268]
**Describe the bug**
I use copyq a lot.  I am a programmer and I quite often use CopyQ as a cut-n-paste store, so the volume of work I give copyq is what I'd call high frequency I don't often have very large things (but sometimes).

This week I was using copy and paste with Mozilla between various web pages and web apps and pdf files via okular.  Not with a code editor.  I am running Linux and the last reboot was around 8 weeks ago.

Tonight copyq failed because there were too many files open.  So even looking at the GUI pop-up gave me an error message and the GUI didn't come up.

I used `ps` to check on use -- There were 331 CopyQ processeses `exec`-ed.  99% of them are: `/usr/bin/copyq eval -- runMenuCommandFilters()` 

**To Reproduce**
1. I don't really know.  Work for many weeks pasting and copying with the browser, etc?
2. Eventually clicking on the paste or copy or show pop-up exhibits an error message from the desktop.
3. I describe this problem as a resource leak

**Expected behavior**
I have not experienced this problem over the last 5 years that I've used CopyQ.  My usual login time is many months.  I feel this is something new from recent updates of a change in copyq

**Screenshots**

N/A

a short version of the 300+ open copyq command may be instructive:

```
 $ ps -ef | grep copyq
william    468 12988  0 Nov27 ?        00:00:00 /usr/bin/copyq eval -- runMenuCommandFilters()
william    476 12988  0 Nov27 ?        00:00:00 /usr/bin/copyq eval -- runMenuCommandFilters()
william    482 12988  0 Nov27 ?        00:00:00 /usr/bin/copyq eval -- runMenuCommandFilters()
william    484 12988  0 Nov27 ?        00:00:00 /usr/bin/copyq eval -- runMenuCommandFilters()
william    490 12988  0 Nov27 ?        00:00:00 /usr/bin/copyq eval -- runMenuCommandFilters()
william    510 12988  0 Nov27 ?        00:00:00 /usr/bin/copyq eval -- runMenuCommandFilters()
   :
 $ ps -ef | grep copyq | wc -l 
331
```

**Version, OS and Environment**

'''
CopyQ Clipboard Manager v3.9.3
Qt: 5.12.2
Compiler: GCC
Arch: x86_64-little_endian-lp64
OS: Ubuntu 19.04
'''

 - Desktop environment, window manager (if applicable)
    - Cinnamon 3.8.8

**Additional context**

Have been heavily using Firefox browser with cut and paste.  I have noticed some applications can influence the copy/past management (e.g Netbeans or other Java apps).

```
 $ /opt/apps/firefox-developer/firefox --version
Mozilla Firefox 71.0b9
```
Also the exec lines are "runMenuCommandFilter".  I don't recognise that function.  I have not been using a search much.  Often I will query for a paste string contents.  I am unclear why that means the process hangs around

**mitigation**

I killed the parent process and restarted CopyQ...

```
ps -ef | grep copyq
william  16792 12944  0 01:09 ?        00:00:00 copyq
william  16804 16792  0 01:09 ?        00:00:00 /usr/bin/copyq --clipboard-access monitorClipboard
william  16810 16792  0 01:09 ?        00:00:00 /usr/bin/copyq eval -- runMenuCommandFilters()
```
Everything looks good.  I anticipate another 6 to 8 weeks before something like this happen again.

Sorry that I can't be more help.  If copyq has a logging facility I can look into that or turn it on, etc.  See if this happens again??

good luck!!

**Comment by hluk** [comment]
You can inspect the logs from main window with F12 shortcut. The log file path is printed with `copyq info log` (it's log-rotated, so the directory could have more log files).

To see more verbose logs run, set `COPYQ_LOG_LEVEL` environment varialbe before lauching the app, e.g. start CopyQ with `COPYQ_LOG_LEVEL=DEBUG copyq`.

**Comment by aplatypus** [comment]
Hi -- F12 didn't work well probably because too many apps want that key for stuff like full-screen, etc.  `Help / Show Log` -- Works fine.

The log view pop-up has different levels shown as Check-Boxes -- That is quite good. Like it!

I have been thinking about this.  I also use CopyQ on my laptop.  That machine hasn't hit this issue and I know that I do maybe use CopyQ 20 x times more there than I do at home.

Looking at the log since I updated/restarted, there is one message that occurs frequently:

```
Warning [2019-12-19 22:42:28.411] <monitorClipboard-16804>: Clipboard data expired, refusing to access old data
Warning [2019-12-19 22:43:39.374] <monitorClipboard-16804>: ELAPSED 679 ms acessing "text/plain"
```

I confess that I don't see what "refusing to access old data" means for a clipboard manager.

Anyway good to know how to use the logging if/when such a problem might resurface.  Thank you.

**Comment by hluk** [comment]
I believe this is issue in some recent Qt framework (quite possibly 5.12.2) - it occasionally took 5s to access X11 clipboard and gave up. There is a workaround for this in CopyQ but it can still happen (there is no easy way to fix it in the app). But I haven't seen this in a while - looks like it's fixed in Qt 5.12.5.

If you can try installing the [flatpak version of CopyQ](https://github.com/hluk/copyq/blob/master/README.md#other-linux-distributions). The most recent version uses Qt 5.12.5.

You can also try disabling image plugin if you don't need CopyQ to handle images in clipboard (you can disable it in Items configuration tab) - there is a known issue with some spreadsheet processors copying all cells as huge images.

**Comment by hluk** [comment]
There should be now only single process `copyq eval -- runDisplayCommands()`.
